### PR TITLE
Implemented more gender options and updated schema.

### PR DESCRIPTION
**Pull request by katrinafyi**
- Store majors as a `|` separated string up to 1000 characters. PostgreSQL does have arrays but I think those would be more unwieldly to use.
- Added non-binary and other options to gender dropdown.
- Other is now stored distinct from unspecified.
- Added optional free text field alongside dropdown (this is not limited to the "other" option).

- **Note:** because of modifications to columns and enum types, this required recreating the table schema.

image

(In exciting news, this is the last of the changes I had planned.)

### CODE REVIEW

**Comment by katrinafyi** [comment]
See discussion in #40.

**Comment by bradleysigma** [comment]
(How exciting.)

**Review by nicklambourne — APPROVED** [review]

**Comment by nicklambourne** [comment]
A couple of suggestions:
- Make the optional extra gender details only appear when 'Other' is selected.
- Include a reason why we collect gender information underneath the input field (e.g. "We collect this information to track the performance of our diversity initiatives."

**Comment by katrinafyi** [comment]
I think the text field should be available to everyone. Even among male, female, and nonbinary, there are different things people might want to write.

I'll add some text giving a reason.

**Review by JamesDearlove — APPROVED** [review]
Looks good to me. As Nick said above, some text saying why we collect gender info would be good.

**Comment by katrinafyi** [comment]
PTAL at changes. Added "We ask for gender information to track our diversity initiatives. Feel free to answer in as much or as little detail as you want." underneath gender fields.

image

File: uqcs/templates/form.mako [hunk 1]
```diff
@@ -77,31 +77,37 @@
...
-            name="gender_text"
-            type="text"
-            class="input"
-            id="gender_text"
-            placeholder="(optional extra details)"
-          />
-        </div>
+        <p class="help">
+          We ask for gender information to track our diversity initiatives.
+          Feel free to answer in as much or as little detail as you want.
```
Comment (nicklambourne) [inline uqcs/templates/form.mako]: This wording feels a tad clunky, maybe:

> Providing this information is completely optional. We gather aggregate gender statistics only to inform and assess our diversity initiatives.

File: uqcs/models.py [hunk 2]
```diff
@@ -69,6 +70,7 @@ class Student(Member):
     domestic = Column(Boolean)
     year = Column(Integer)
     program = Column(String(100))
+    majors = Column(String(1000))
```
Comment (TRManderson) [inline uqcs/models.py]: Tbh you basically don't have to ever bother providing a length in postgres, `Text` alone is always good enough

File: uqcs/app.py [hunk 3]
```diff
@@ -77,6 +76,8 @@ def user_from_request(form):
             info["undergrad"] = form['degreeType'].lower() == 'undergrad'
         if 'degree' in form:
             info['program'] = form['degree'][:100]
+        if 'majors' in form:
+            info['majors'] = '|'.join(form.getlist('majors'))[:1000]
```
Comment (TRManderson) [inline uqcs/app.py]: sqlalchemy is magic and postgres arrays are actually pretty chill if you wanna do that instead - you can literally just pass the list to the constructor

https://docs.sqlalchemy.org/en/13/core/type_basics.html#sqlalchemy.types.ARRAY
https://docs.sqlalchemy.org/en/13/dialects/postgresql.html#sqlalchemy.dialects.postgresql.ARRAY

**Review by TRManderson — APPROVED** [review]
LGTM, you don't have to implement any of my changes but you can if you want

File: uqcs/app.py (hunk 3 above)
Comment (katrinafyi) [reply uqcs/app.py]: It is indeed ✨magic ✨

**Pull request merged**